tests/B4RealTest.spec.ts

```diff
@@ -136,11 +165,172 @@ describe("B4REAL token tests", async () => {
 
     const currentBalanceEth = Number(ethers.utils.formatEther(currentBalance));
 
-    // 2% of 100 is 2, 100 - 2 = 98 ect
     expect(taxBalance.toString()).to.equal("10");
     expect(currentBalanceEth.toString()).to.equal("90");
   });
 
+  it("Should test the transferFrom function and not tax with a normal tranaction", async () => {
+    const openingBalance = await B4REAL.balanceOf(secondUserAddress);
+
+    const openingTaxBalance = Number(
+      ethers.utils.formatEther(await B4REAL.balanceOf(taxAddress))
+    );
+
+    // Nothing in this account yet
+    expect(openingBalance.toString()).to.equal("0");
+
+    const amount = ethers.utils.parseEther("1");
+
+    await hre.network.provider.request({
+      method: "hardhat_impersonateAccount",
+      params: [userAddress]
+    });
+
+    // Approve secondUserAddress
+    await B4REAL.connect(userSigner).approve(secondUserAddress, amount);
+
+    const allowance = await B4REAL.allowance(userAddress, secondUserAddress);
+    // Check allowance is correct
+    expect(allowance).to.equal(amount.toString());
+    await hre.network.provider.request({
+      method: "hardhat_stopImpersonatingAccount",
+      params: [userAddress]
+    });
+
+    await hre.network.provider.request({
+      method: "hardhat_impersonateAccount",
+      params: [secondUserAddress]
+    });
+
+    await B4REAL.connect(secondUserSigner).transferFrom(
+      userAddress,
+      secondUserAddress,
+      amount
+    );
+    await hre.network.provider.request({
+      method: "hardhat_stopImpersonatingAccount",
+      params: [secondUserAddress]
+    });
+
+    const currentUser2Balance = await B4REAL.balanceOf(secondUserAddress);
+
+    const closingTaxBalance = Number(
+      ethers.utils.formatEther(await B4REAL.balanceOf(taxAddress))
+    );
+
+    // no tax should be applied so full amount should be sent to secondUserAddress
+    // and the tax address should have no balance change
+    expect(currentUser2Balance.toString()).to.equal(amount.toString());
+    expect(openingTaxBalance.toString()).to.equal(closingTaxBalance.toString());
+  });
+
+  it("Should test the transferFrom function and apply tax correctly", async () => {
+    await B4REAL.includeInFee(secondUserAddress);
+    const isWhitelisted = await B4REAL.whitelisted(secondUserAddress);
+    expect(isWhitelisted).to.equal(true);
+
+    const openingBalance = await B4REAL.balanceOf(secondUserAddress);
+
+    // Just the tokens from the previous test in this account
+    expect(openingBalance.toString()).to.equal(ethers.utils.parseEther("1"));
```

Comment:
Big Number assert?

Reply:
See my comment above about `ethereum-waffle` middleware for how to do that.

Reply:
Could do
```
expect(openingBalance.eq(ethers.utils.parseEther("1"))).to.be.true;
```